src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -165,13 +175,43 @@ private Task StartAsyncCore()
                         return;
                     }
 
-                    _transport = _transportFactory.CreateTransport(GetAvailableServerTransports(negotiationResponse));
-                    connectUrl = CreateConnectUrl(Url, negotiationResponse);
-                }
+                    // This should only need to happen once
+                    _serverTransports = GetAvailableServerTransports(negotiationResponse);
+                    connectUrl = CreateConnectUrl(Url, negotiationResponse.ConnectionId);
+
+                    foreach (var transport in AllTransports)
+                    {
+                        try
+                        {
+                            if ((transport & _serverTransports & _requestedTransportType) != 0)
+                            {
+                                // The negotiation response gets cleared in the fallback scenario.
+                                if (negotiationResponse == null)
+                                {
+                                    negotiationResponse = await GetNewNegotiationResponse();
+                                    connectUrl = CreateConnectUrl(Url, negotiationResponse.ConnectionId);
+                                }
 
-                _logger.StartingTransport(_transport, connectUrl);
-                await StartTransport(connectUrl);
+                                await StartTransport(connectUrl, transport);
+                                _logger.StartingTransport(_transport, connectUrl);
+                                break;
+                            }
+                        }
+                        catch (Exception ex)
+                        {
+                            // Try the next transport
+                            _logger.TransportFailedToStart(nameof(transport), ex);
```

Comment:
Why are we using nameof(transport), won't that just end up being "transport"? You want the transport.ToString() right?